src/main/scala/io/scalac/wtf/domain/UserService.scala

```diff
@@ -0,0 +1,36 @@
+package io.scalac.wtf.domain
+
+import cats.data.Validated.{Invalid, Valid}
+import cats.{Applicative, Functor, Monad}
+import cats.implicits._
+import slick.dbio.DBIO
+import slick.driver.H2Driver.api._
+
+import scala.concurrent.ExecutionContext
+
+object UserService {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  /*
+  //TODO: Move this somewhere else
+  //Evidence for DBIO being a cats Monad
+  implicit val dbioMonad = new Functor[DBIO] with Applicative[DBIO] with Monad[DBIO] {
+    def pure[A](a: A): DBIO[A] = DBIO.successful(a)
+    def flatMap[A, B](fa: DBIO[A])(f: A => DBIO[B]) = fa.flatMap(f)
+  }
+*/
+  def createUser(createdUser: User)(implicit executionContext: ExecutionContext): DBIO[UserId] = {
+    UserRepository.findByEmail(createdUser.email).flatMap {
+      case Some(_) => DBIO.failed(new Exception("Provided email is already taken"))
```

Comment:
I had doubts about it too but a guy explained this will not be thrown but just signalizes the Future[T] returned by db.run(...) on this will finish as a Failure (which is a subtype of Future). This is just a value (nothing is thrown) that signalizes DBIOAction resulted in a Failure or Succes (both are subtypes of Future)